### ISSUE TITLE
https://tiktok-tts.weilnet.workers.dev/api/generation is returning 500

### ISSUE THREAD

**Issue by cunhar** [Weilbyte/tiktok-tts#54]

**Comment by cunhar** [comment]
```
{
    "success": false,
    "data": null,
    "error": "Couldn’t load speech. Try again."
}
```

**Comment by cunhar** [comment]
@Weilbyte are you around?

**Comment by Weilbyte** [comment]
What is the request you are sending?

**Comment by cunhar** [comment]
```
{
    "text": "Fighters, prepare to face Bones in battle!",
    "voice": "en_us_006"
}
```

**Comment by cunhar** [comment]
{
    "text": "Fighters, prepare to face Bones in battle!",
    "voice": "en_us_006"
}

**Comment by cunhar** [comment]
your main site is working fine, but your workers.dev and other sites as well are down with similar messages  i tried to spin up my own based on another project and also get an error with the tiktok api -- @Weilbyte

**Comment by Xkipper** [comment]
```console
POST https://tiktok-tts.weilnet.workers.dev/api/generation 500 (Internal Server Error)
```

```JSON
{
	text: 'boom boom',
	voice: 'es_mx_002'
}
```

**Comment by Letsbeginner** [comment]
Confirmed. I am getting the 500 error. Please assist.
       
       POST https://tiktok-tts.weilnet.workers.dev/api/generation 500 (Internal Server Error)

**Comment by cunhar** [comment]
Looks like the session id is now associated with an endpoint

**Comment by Letsbeginner** [comment]
Can you explain? is it working on your end now?

**Comment by cunhar** [comment]
No, it's not.  I ran my own tests locally and noticed the issue.  The
maintainer will probably address it soon in this project

**Comment by Weilbyte** [comment]
It seems to be an issue with how the framework used for the Worker HTTP server is parsing the request body. @cunhar how exactly are you sending the requests (e.g. directly with cURL)?

**Comment by cunhar** [comment]
I use a JavaScript project

**Comment by Weilbyte** [comment]
Can you share me the part of the code that sends the request?

**Comment by Weilbyte** [comment]
For example, this code works for me:
```js
const url = 'https://tiktok-tts.weilnet.workers.dev/api/generation';
const options = {
  method: 'POST',
  headers: {'content-type': 'application/json'},
  body: '{"text":"Hello","voice":"en_us_001"}'
};

try {
  const response = await fetch(url, options);
  const data = await response.json();
  console.log(data);
} catch (error) {
  console.error(error);
}
```

**Comment by cunhar** [comment]
It's very similar.  The endpoints were not responding with the sound data.
It was saying a generic 500

**Comment by Xkipper** [comment]
> For example, this code works for me:
> 
> ```js
> const url = 'https://tiktok-tts.weilnet.workers.dev/api/generation';
> const options = {
>   method: 'POST',
>   headers: {'content-type': 'application/json'},
>   body: '{"text":"Hello","voice":"en_us_001"}'
> };
> 
> try {
>   const response = await fetch(url, options);
>   const data = await response.json();
>   console.log(data);
> } catch (error) {
>   console.error(error);
> }
> ```

Now work. Thanks You <3

**Comment by Letsbeginner** [comment]
I am pretty sure this is solved, hallelujah!
It seems to have been something on Weilbytes end, because I had the same error, but my code wasn't changed. I sent my request based on the instructions. And now, its working and my code wasn't changed. It fixed itself on its own; however, I am still curious to what it was.
